src/gallium/drivers/lima/lima_bo.c

```diff
@@ -130,8 +130,10 @@ void lima_bo_free(struct lima_bo *bo)
       util_hash_table_remove(screen->bo_flink_names, (void *)bo->flink_name);
    mtx_unlock(&screen->bo_table_lock);
 
-   if (bo->va)
+   if (bo->va) {
+      lima_bo_va_unmap(bo, bo->va);
```

Comment:
Strange, it should be done in the bo free function in kernel:
lima_gem_free_object

Can you see why it's not va unmap there?

Reply:
My guess is that lima_gem_free_object() hasn't been called yet, probably something's keeping a reference for it. Or we may be leaking a bo.

Anyway, if bo is still alive we shouldn't reuse it's va. So it's better to unmap va before we lost track of bo in userspace.

Reply:
Right, we should do it here. But I'm afraid there's bo leaking too because no other place in current use case should hold another reference of this bo's gem object.

Reply:
@yuq can you try running glmark2-es without this patch to see whether the issue is present in 4.13 kernel? I'm running 4.15-rc5, so I could make a mistake while porting your patches to 4.15.

Reply:
OK, I'll try on my side too.

Reply:
I found this bo is exported to display drm driver, so not freed. In this case we indeed have to va_unmap it explicitly here. Thanks.